eng/_core/cmd/build/build.go

```diff
@@ -241,15 +241,43 @@ func runCmd(cmd *exec.Cmd) error {
 	return cmd.Run()
 }
 
+func retry(attempts int, f func() error) error {
+	var i = 0
+	for ; i < attempts; i++ {
+		if attempts > 1 {
+			fmt.Printf("---- Running attempt %v of %v...\n", i+1, attempts)
+		}
+		err := f()
+		if err != nil {
+			if i+1 < attempts {
+				fmt.Printf("---- Attempt failed with error: %v\n", err)
+				continue
+			}
+			fmt.Printf("---- Final attempt failed.\n")
+			return err
+		}
+		break
+	}
+	fmt.Printf("---- Successful on attempt %v of %v.\n", i+1, attempts)
+	return nil
+}
+
 func getMaxMakeRetryAttempts() (int, error) {
-	const retryEnvVarName = "GO_MAKE_MAX_RETRY_ATTEMPTS"
-	a := os.Getenv(retryEnvVarName)
+	return getEnvIntOrDefault("GO_MAKE_MAX_RETRY_ATTEMPTS", 1)
+}
+
+func getMaxTestRetryAttempts() (int, error) {
+	return getEnvIntOrDefault("GO_TEST_MAX_RETRY_ATTEMPTS", 1)
+}
+
+func getEnvIntOrDefault(varName string, defaultValue int) (int, error) {
+	a := os.Getenv(varName)
 	if a == "" {
```

Comment:
Better use [os.LookupEnv](https://pkg.go.dev/os#LookupEnv). It covers the case where the user has incorrectly set the environment variable to an empty string.

Reply:
This depends on what we want the behavior to be, though, right? I was intending to treat unset and empty string both as default, because in my experience, people aren't particularly familiar with `x=` vs. `unset x`.

But... rejecting empty string *could* help if you're trying to set it like `GO_MAKE_MAX_RETRY_ATTEMPTS=$(fun)`, and for some reason `fun` fails rather than sending anything to stdout.

I'll switch over and write an error message for empty string that should point people in the right direction.